src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -62,6 +73,25 @@ trait RegularSync {
       }
   }
 
+  private def insertMinedBlock(block: Block, parentTd: BigInt) = {
+    val result: Either[BlockExecutionError, Seq[Receipt]] = ledger.executeBlock(block, blockchainStorages, validators)
```

Comment:
If I'm not mistaken, we are executing the block twice. First when preparing the block for mining, then here. Can we do that only when preparing the block?